salt/cloud/__init__.py

```diff
@@ -1454,10 +1454,13 @@ def do_action(self, names, kwargs):
                     if not names:
                         break
                     if vm_name not in names:
-                        log.debug('vm:{0} in provider:{1} is not in name list:{2!r}'.format(
-                            vm_name, driver, names
-                        ))
-                        continue
+                        if 'id' in vm_details and vm_details['id'] in names:
+                            vm_name = vm_details['id']
+                        else:
+                            log.debug('vm:{0} in provider:{1} is not in name list:{2!r}'.format(
```

Comment:
The log message should be something like "<name> is not in the names list" or something similar.